### PR TITLE
Update Dockerfile.skema-py

### PR DESCRIPTION
**Pull request by enoriega**
Added a `pip install` dependency required to support running the TR notebook

### CODE REVIEW

**Comment by myedibleenso** [comment]
@enoriega , this can't be included for the time being (next couple of months), as your library depends on Python >=3.9.  `skema` is currently stuck on v3.8 until we move to using `tree-sitter` to process Python source code for program analysis.

Once that change has been made, it would be preferable to add this dependency under [the `tr` optional dependencies in the `pyproject.toml`](https://github.com/ml4ai/skema/blob/9b7c26c165c149254901cf1feba089b54a0b458f/pyproject.toml#L57) (ex something like `"askem_extractions[all]@git+https://github.com/ml4ai/ASKEM-TA1-DataModel"`)

**Comment by myedibleenso** [comment]
For the time being, I suggest using a separate example subdirectory with its own docker-compose.yml file specific to text reading in the DockerVM repo.  Will that work?

**Comment by enoriega** [comment]
Yeah. I think this works for tomorrow. The code should run with python 3.8, albeit a few minor type hint changes. I will make sure it works and follow your advice on the dependency install

**Comment by myedibleenso** [comment]
[Here is a recap of the discussion from 4/6/2023 on Slack about this](https://mlbayes.slack.com/archives/CF1S834KH/p1680836252866099?thread_ts=1680820645.268039&cid=CF1S834KH):

>GHP: I noticed in the pyproject.toml that the python library does not currently work with Python versions > 3.8 (presumably due to something about the program_analysis package).  Does this relate to a particular dependency?  I don’t see any specific deps (https://github.com/ml4ai/skema/blob/84f804005f3227f4826c67cbdbe6933517273dfe/pyproject.toml#L12-L16) that would make me think so.  I ask partly because I was able to install everything using 3.9 (after changing the toml) and run the tests (two failing).

>CM: There is a current dependency of the PA pipeline on Python 3.8 b/c it depends on the python native ast. The ast, unfortunately, does not guarantee backward compatibility, so changes in 3.9 break things. This is something that in the future I hope we can remove, perhaps by migrating the python front-end to use tree-sitter’s python parser. But right now there are higher priorities.

**Comment by enoriega** [comment]
Right. `gensim` was the culprit that pinned me to 3.9 at least for TR. Now that I am reworking the gromet aligner, we are replacing gensim with huggingface and I think it should work. So maybe soon I can make all the TR python code compatible with 3.8.

**Review by myedibleenso — CHANGES_REQUESTED** [review]
- remove TR util library's dependency on Python v3.9
- add dependency under `tr` optionals in `pyproject.toml` (instead of listing in Dockerfile)

File: pyproject.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ authors = [
 description = "Scientific Knowledge Extraction and Model Analysis"
 version = "0.1.0"
 dependencies=[
-    "gensim",
+    #"gensim",
```
Comment (myedibleenso) [inline pyproject.toml:14]: Let's just remove this.

File: pyproject.toml [hunk 2]
```diff
@@ -54,7 +54,7 @@ isa = [
 img2mml = ["fastapi", "uvicorn", "torch", "torchvision", "python-multipart"]
 
 # dependencies for text reading utilities.
-tr = ["transformers", "gensim", "SPARQLWrapper", "tqdm", "pandas"]
+tr = ["transformers", "SPARQLWrapper", "tqdm", "pandas", "askem_extractions[all]@git+https://github.com/ml4ai/ASKEM-TA1-DataModel"]
```
Comment (myedibleenso) [inline pyproject.toml:57]: Awesome!  Have you confirmed (ex. from the REPL) this pulls in your dep?

**Review by myedibleenso — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (enoriega) [reply pyproject.toml:56]: Yes, it works. It builds and I am able to import the code from within a python session inside the container

**Review by myedibleenso — APPROVED** [review]

**Pull request merged**